youtube_dl/extractor/americastestkitchen.py

```diff
@@ -90,3 +92,74 @@ def _real_extract(self, url):
             'series': try_get(episode, lambda x: x['show']['title']),
             'episode': episode.get('title'),
         }
+
+
+class AmericasTestKitchenSeasonIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?(?P<show>americastestkitchen|cookscountry)\.com/episodes/browse/season\_(?P<id>\d+)'
+    _TESTS = [{
+        # ATK Season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_1',
+        'info_dict': {
+            'id': 'season-1',
+            'title': 'Season 1',
+        },
+        'playlist_count': 13,
+    }, {
+        # Cooks Country Season
+        'url': 'https://www.cookscountry.com/episodes/browse/season_12',
+        'info_dict': {
+            'id': 'season-12',
+            'title': 'Season 12',
+        },
+        'playlist_count': 13,
+    }, {
+        # Multi-digit season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_20',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        show_name, season = re.match(self._VALID_URL, url).groups()
+
+        slug = 'atk' if show_name == 'americastestkitchen' else 'cco'
+
+        filters = [
+            'search_season_list:Season %s' % season,
+            'search_document_klass:episode',
+            'search_show_slug:%s' % slug,
+        ]
+
+        params = {
+            'facetFilters': json.dumps(filters),
+            'attributesToRetrieve': 'search_url',
+            'attributesToHighlight': '',
+            # ATK and CCO always have less 26 episodes per season
+            'hitsPerPage': '30',
+        }
+
+        url = 'https://y1fnzxui30-dsn.algolia.net/1/indexes/everest_search_atk_season_desc_production'
+
+        season_search = self._download_json(
+            update_url_query(url, params),
+            season,
+            headers={
+                'Origin': 'https://www.americastestkitchen.com',
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'X-Algolia-API-Key': '8d504d0099ed27c1b73708d22871d805',
+                'X-Algolia-Application-Id': 'Y1FNZXUI30',
+            })
+
+        entries = [
+            self.url_result(
+                'https://www.americastestkitchen.com' + episode['search_url'],
+                'AmericasTestKitchen',
+                episode['objectID'].split('_')[-1])
+            for episode in season_search['hits']
+        ]
```

Comment:
extraction should not break if an episode or all episodes couldn't be extracted.